### PR TITLE
Remove trailing zeros from the output of BytesUtils::formatBytes

### LINKED ISSUE #7198: [isoltest] String formatting should not output padded zeros
Sneaked in with https://github.com/ethereum/solidity/pull/7080.

The following test
```
contract C {
    function e() public {
        revert("Transaction failed.");
    }
}
// ====
// EVMVersion: >homestead
// ----
// _() -> FAILURE
// e() ->
```
does not print strings correctly:
```
Obtained result:
// _() -> FAILURE
// e() -> FAILURE, hex"08c379a0", 0x20, 19, "Transaction failed.\0\0\0\0\0\0\0\0\0\0\0\0\0"
```
It should rather output
```
// e() -> FAILURE, hex"08c379a0", 0x20, 19, "Transaction failed."
```

### PR DESCRIPTION
**Pull request by ghallak**
Closes #7198

### CODE REVIEW

File: test/libsolidity/util/TestFunctionCall.h [hunk 1]
```diff
@@ -50,7 +50,7 @@ class TestFunctionCall
 	TestFunctionCall(FunctionCall _call): m_call(std::move(_call)) {}
 
 	/// Formats this function call test and applies the format that was detected during parsing.
-	/// If _renderResult is false, the expected result of the call will is used, if it's false
+	/// If _renderResult is false, the expected result of the call will be used, if it's true
```
Comment (erak) [inline test/libsolidity/util/TestFunctionCall.h:53]: Oops, thanks!

File: test/libsolidity/util/BytesUtils.cpp [hunk 2]
```diff
@@ -263,8 +263,15 @@ string BytesUtils::formatBytes(
 		os << formatHexString(_bytes);
 		break;
 	case ABIType::String:
-		os << formatString(_bytes);
+	{
+		auto paddingZerosCnt = find_if(
```
Comment (erak) [inline test/libsolidity/util/BytesUtils.cpp]: I think it would be better to move this into its own function and then call it from here.

**Comment by erak** [comment]
Thanks for submitting this PR. In general it looks good but I still left a tiny request for change.

File: test/libsolidity/util/BytesUtils.cpp (hunk 2 above)
Comment (erak) [reply test/libsolidity/util/BytesUtils.cpp]: Maybe `BytesUtils::countRightPaddedZeros`?

**Review by erak — CHANGES_REQUESTED** [review]
Thanks for submitting this PR. In general it looks good but I still left a tiny request for change.

File: test/libsolidity/util/BytesUtils.cpp (hunk 2 above)
Comment (ghallak) [reply test/libsolidity/util/BytesUtils.cpp]: Done that here 8a8f553e4d1ada792bb31f1b15df3675f4cc7f64

**Review by erak — APPROVED** [review]

**Pull request merged**